scripts/ci/pre_commit/compile_ui_assets.py

```diff
@@ -68,10 +70,19 @@ def get_directory_hash(directory: Path, skip_path_regexp: str | None = None) ->
         shutil.rmtree(dist_directory, ignore_errors=True)
     env = os.environ.copy()
     env["FORCE_COLOR"] = "true"
-    subprocess.check_call(
-        ["pnpm", "install", "--frozen-lockfile", "--config.confirmModulesPurge=false"],
-        cwd=os.fspath(ui_directory),
-    )
+    for try_num in range(3):
+        print(f"### Trying to install yarn dependencies: attempt: {try_num} ###")
```

Comment:
Nit: Let's move this entire block into a function? We can reuse that time. And, we will be able to skip the try_number==2 check too

Reply:
Not sure if I understand :) . .. These two are separate scripts - we currently don't read anything from "common" scripts in them, that would be somewhat increasing coupling between them.  I think I prefere to not apply DRY here.

Reply:
Especially that one of them will go away when we remove old UI :)

Reply:
Ah yes, good point. Lets keep it then